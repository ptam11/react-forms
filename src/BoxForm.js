import React, { Component } from 'react';

class BoxForm extends Component {
  constructor(props) {
    super(props);
    this.state= {
      width: '',
      height: '',
      backgroundColor: ''
    }
    this.handleChange = this.handleChange.bind(this);
    this.submitForm = this.submitForm.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  submitForm(e) {
    e.preventDefault();
    this.props.addBox(this.state);
    this.setState({
      width: '',
      height: '',
      backgroundColor: ''
    })
  }
  
  render() {
    return (
      <div>
        <form onSubmit={this.submitForm} >
          <label htmlFor="width" > Width: </label>
          <input onChange={this.handleChange} value={this.state.width} type='number' name="width" />
          <label htmlFor="height" > Height: </label>
          <input onChange={this.handleChange} value={this.state.height} type='number' name="height" />
          <label htmlFor="backgroundColor" > Background Color: </label>
          <input onChange={this.handleChange} value={this.state.backgroundColor} type='text' name="backgroundColor" />
          <button>Add Box!!!!!</button>
        </form>
      </div>
    );
  }

}

export default BoxForm;
