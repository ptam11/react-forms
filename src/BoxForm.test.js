import React from 'react';
import BoxForm from './BoxForm';
import { shallow, mount, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({
  adapter: new Adapter()
});

it('renders without errors', () => {
  shallow(<BoxForm />);
});

it('snapshot matches', () => {
  let wrapper = shallow(<BoxForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

xit('it renders initial message', () => {
  let wrapper = mount(<BoxForm />);
  expect(wrapper.exists('p')).toEqual(true);
  expect(wrapper.html()).toContain("Think of a Question");
});

xit('it renders correct message', () => {
  let wrapper = mount(<BoxForm />);
  wrapper.setState({ msg: "Hello", color: "rebeccapurple" });
  let p_elem = wrapper.find('p').first();
  expect(p_elem.equals(<p>Hello</p>)).toEqual(true);
});

const shortList = [
  { msg: "Whiskey is a dog.", color: "green" },
  { msg: "Whiskey likes food.", color: "green" },
  { msg: "Whiskey likes sleeping.", color: "green" },
]

xit('it changes default message on click', () => {
  let wrapper = mount(<BoxForm  answers={shortList}/>);
  wrapper.simulate("click");
  expect(wrapper.html()).toContain("Whiskey");
});