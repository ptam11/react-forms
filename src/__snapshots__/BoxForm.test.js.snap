// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="width"
    >
       Width: 
    </label>
    <input
      name="width"
      onChange={[Function]}
      type="number"
      value=""
    />
    <label
      htmlFor="height"
    >
       Height: 
    </label>
    <input
      name="height"
      onChange={[Function]}
      type="number"
      value=""
    />
    <label
      htmlFor="backgroundColor"
    >
       Background Color: 
    </label>
    <input
      name="backgroundColor"
      onChange={[Function]}
      type="color"
      value="#000000"
    />
    <button>
      Add Box!!!!!
    </button>
  </form>
</div>
`;
